### PR TITLE
Remove dependency on pytest-dbfixtures

### PR DESCRIPTION
**Pull request by bmoscon**
Start using pytest-server-fixtures

### CODE REVIEW

**Comment by bmoscon** [comment]
@jamesblackburn

**Comment by bmoscon** [comment]
@eeaston any idea why i keep seeing this failure on 3.4? 

```
__ ERROR at setup of test_should_return_data_when_date_range_spans_libraries ___
mongo_server = <pytest_server_fixtures.mongo.MongoTestServer object at 0x7fb2b65c7198>
    @pytest.fixture(scope="function")
    def arctic(mongo_server):
        logger.info('arctic.fixtures: arctic init()')
>       mongo_server.api.drop_database('arctic')
arctic/fixtures/arctic.py:22: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../virtualenv/python3.4.4/lib/python3.4/site-packages/pymongo/mongo_client.py:1112: in drop_database
    read_preference=ReadPreference.PRIMARY)
../../../virtualenv/python3.4.4/lib/python3.4/site-packages/pymongo/database.py:478: in command
    with client._socket_for_reads(read_preference) as (sock_info, slave_ok):
/opt/python/3.4.4/lib/python3.4/contextlib.py:59: in __enter__
    return next(self.gen)
../../../virtualenv/python3.4.4/lib/python3.4/site-packages/pymongo/mongo_client.py:798: in _socket_for_reads
    with self._get_socket(read_preference) as sock_info:
/opt/python/3.4.4/lib/python3.4/contextlib.py:59: in __enter__
    return next(self.gen)
../../../virtualenv/python3.4.4/lib/python3.4/site-packages/pymongo/mongo_client.py:762: in _get_socket
    server = self._get_topology().select_server(selector)
../../../virtualenv/python3.4.4/lib/python3.4/site-packages/pymongo/topology.py:210: in select_server
    address))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <pymongo.topology.Topology object at 0x7fb2b04a6a58>
selector = Primary(), server_selection_timeout = None, address = None
    def select_servers(self,
                       selector,
                       server_selection_timeout=None,
                       address=None):
        """Return a list of Servers matching selector, or time out.
    
            :Parameters:
              - `selector`: function that takes a list of Servers and returns
                a subset of them.
              - `server_selection_timeout` (optional): maximum seconds to wait.
                If not provided, the default value common.SERVER_SELECTION_TIMEOUT
                is used.
              - `address`: optional server address to select.
    
            Calls self.open() if needed.
    
            Raises exc:`ServerSelectionTimeoutError` after
            `server_selection_timeout` if no matching servers are found.
            """
        if server_selection_timeout is None:
            server_timeout = self._settings.server_selection_timeout
        else:
            server_timeout = server_selection_timeout
    
        with self._lock:
            self._description.check_compatible()
    
            now = _time()
            end_time = now + server_timeout
            server_descriptions = self._apply_selector(selector, address)
    
            while not server_descriptions:
                # No suitable servers.
                if server_timeout == 0 or now > end_time:
                    raise ServerSelectionTimeoutError(
>                       self._error_message(selector))
E                   pymongo.errors.ServerSelectionTimeoutError: testing-worker-linux-docker-641955fd-3477-linux-14:54694: [Errno 111] Connection refused
../../../virtualenv/python3.4.4/lib/python3.4/site-packages/pymongo/topology.py:186: ServerSelectionTimeoutError
```

File: arctic/fixtures/arctic.py [hunk 1]
```diff
@@ -1,55 +1,47 @@
...
 
 
 @pytest.fixture(scope="function")
-def arctic(mongodb):
+def arctic(mongo_server):
     logger.info('arctic.fixtures: arctic init()')
-    mongodb.drop_database('arctic')
-    mongodb.drop_database('arctic_{}'.format(getpass.getuser()))
-    arctic = m.Arctic(mongo_host=mongodb)
+    retry_count = 0
```
Comment (jamesblackburn) [inline arctic/fixtures/arctic.py]: We shouldn't need this, should we?

Comment (bmoscon) [inline arctic/fixtures/arctic.py]: correct, we're going to put a fix in the fixtures code (I believe)

**Pull request merged**